terraform/eks/main.tf

```diff
@@ -344,13 +398,249 @@ resource "kubernetes_deployment" "sample_app_deployment" {
+    namespace = kubernetes_namespace.aoc_ns.metadata[0].name
+    labels = {
+      app = "aoc"
+    }
+  }
+
+  spec {
+    replicas = 1
+
+    selector {
+      match_labels = {
+        app = "aoc"
+      }
+    }
+
+    template {
+      metadata {
+        labels = {
+          app = "aoc"
+        }
+      }
+
+      spec {
+        service_account_name = "aoc-role"
+
+        volume {
+          name = "otel-config"
+          config_map {
+            name = kubernetes_config_map.aoc_config_map.metadata[0].name
+          }
+        }
+
+        volume {
+          name = "mocked-server-cert"
+          config_map {
+            name = kubernetes_config_map.mocked_server_cert.metadata[0].name
+          }
+        }
+
+        volume {
+          name = kubernetes_service_account.aoc-role.default_secret_name
+          secret {
+            secret_name = kubernetes_service_account.aoc-role.default_secret_name
+          }
+        }
+
+        container {
+          name = "mocked-server"
+          image = local.mocked_server_image
+
+          readiness_probe {
+            http_get {
+              path = "/"
+              port = 8080
+            }
+            initial_delay_seconds = 10
+            period_seconds = 5
+          }
+        }
+
+        # aoc
+        container {
+          name = "aoc"
+          image = module.common.aoc_image
+          image_pull_policy = "Always"
+          args = [
+            "--config=/aoc/aoc-config.yml"]
+
+          env {
+            name = "SAMPLE_APP_HOST"
+            value = kubernetes_service.pull_mode_sample_app_service[0].load_balancer_ingress.0.hostname
+          }
+
+          env {
+            name = "SAMPLE_APP_PORT"
+            value = module.common.sample_app_lb_port
+          }
+
+          resources {
+            requests {
+              cpu = "0.2"
+              memory = "256Mi"
+            }
+          }
+
+          volume_mount {
+            mount_path = "/var/run/secrets/kubernetes.io/serviceaccount"
+            name = kubernetes_service_account.aoc-role.default_secret_name
+            read_only = true
+          }
+
+          volume_mount {
+            mount_path = "/aoc"
+            name = "otel-config"
+          }
+
+          volume_mount {
+            mount_path = "/etc/pki/tls/certs"
+            name = "mocked-server-cert"
+          }
+        }
+      }
+    }
+  }
+}
+
+# create service upon the mocked server
+resource "kubernetes_service" "pull_mode_mocked_server_service" {
+  count = var.sample_app_mode == "pull" ? 1 : 0
+
+  metadata {
+    name = "mocked-server"
+    namespace = kubernetes_namespace.aoc_ns.metadata[0].name
+  }
+  spec {
+    selector = {
+      app = kubernetes_deployment.pull_mode_aoc_deployment[0].metadata[0].labels.app
+    }
+
+    type = "LoadBalancer"
+
+    port {
+      port = 80
+      target_port = 8080
+    }
+  }
+}
+
+# deploy sample app
+resource "kubernetes_deployment" "pull_mode_sample_app_deployment" {
```

Comment:
since this file is going to be too long, can we put all the things about pull mode into a separate file named pull_mode_deployment.tf something?